addon/models/translation.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2015, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+
+import EmberObject from '@ember/object';
+
+const hasOwnProperty = Object.prototype.hasOwnProperty;
+
+function flatten(translations: Map<string, string>, src: any, prefix?: string) {
+  for (const key in src) {
+    if (!hasOwnProperty.call(src, key)) {
+      continue;
+    }
+
+    const value = src[key];
+    const tKey = prefix ? `${prefix}.${key}` : key;
+
+    if (typeof value === 'object' && value) {
+      flatten(translations, value, tKey);
+    } else {
+      translations.set(tKey, value);
+    }
+  }
+}
+
+// export as named and default so the default adapter can augment the class types because default exports cannot be augmented
+// see http://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation
+export class TranslationModel extends EmberObject {
+  translations: Map<string, string> = new Map();
+
+  init() {
```

Comment:
It's my understanding that the declaration (`translations: Map<string, string> = new Map();`) should have initialized translations in the constructor. For some reason in older ember versions that was not the case, so I initialize in `init` if it's missing. I can investigate wether manually initializing in the constructor instead of `init` works in the older versions.

Reply:
The `ember-lts-2.16` and `ember-lts-2.18` scenarios fail when initializing in the constructor instead of `init`.